README.md

```diff
@@ -766,6 +766,43 @@ nestingLimit: 6
 
 This will prevent the user from nesting tags more than 6 levels deep. Tags deeper than that are stripped out exactly as if they were disallowed. Note that this means text is preserved in the usual ways where appropriate.
 
+### Advanced filtering
+
+For more advanced filtering you can hook directly into the parsing process using three events:
+- `onOpenTag(name, attribs)`: Triggered when an opening tag is encountered.
+- `onText(text)`: Triggered when a text node is encountered.
+- `onCloseTag(name, isImplied)`: Triggered when a closing tag is encountered.
+
+For example, you may want to add spaces around a removed tag, like this:
+```js
+const allowedTags = [ 'b' ];
+let addSpace = false;
+const sanitizedHtml = sanitizeHtml(
+  'Text&#39;s here<div>it&#39;s here</div><div><p>it&#39;s there</p></div>and <b>also</b> here.',
```

Comment:
I would prefer not to have HTML entities in this example. It makes it harder to read and understand.

Reply:
Got it. Will simplify. I added them deliberately because when they are present `htmlparser2` seems to produce two text nodes in a row, and we need to make sure no extra space is inserted before the second text node. But maybe for this simple example no need to overthink it.